loopchain/baseservice/tx_message.py

```diff
@@ -0,0 +1,106 @@
+"""helper class for TxMessages"""
+
+import json
+import queue
+import threading
+
+import sys
+from typing import List, Optional
+
+from loopchain import configure
+from loopchain.blockchain.transactions import Transaction, TransactionVersioner, TransactionSerializer
+from loopchain.protos import loopchain_pb2
+
+
+class TxItem:
+    tx_serializers = {}
+
+    def __init__(self, tx_json: str, channel: str):
+        self.channel = channel
+        self.__tx_json = tx_json
+        self.__len = sys.getsizeof(tx_json) + sys.getsizeof(channel)
+
+    def __len__(self):
+        return self.__len
+
+    def get_tx_message(self):
+        message = loopchain_pb2.TxSend(
+            tx_json=self.__tx_json,
+            channel=self.channel)
+        return message
+
+    @classmethod
+    def create_tx_item(cls, tx_param: tuple, channel: str):
+        tx, tx_versioner = tx_param
+        tx_serializer = cls.get_serializer(tx, tx_versioner)
+        tx_item = TxItem(
+            json.dumps(tx_serializer.to_raw_data(tx)),
+            channel
+        )
+        return tx_item
+
+    @classmethod
+    def get_serializer(cls, tx: Transaction, tx_versioner: TransactionVersioner):
+        if tx.version not in cls.tx_serializers:
+            cls.tx_serializers[tx.version] = TransactionSerializer.new(tx.version, tx.type(), tx_versioner)
+        return cls.tx_serializers[tx.version]
+
+
+class TxMessages:
+    def __init__(self):
+        self._total_size: int = 0
+        self._messages: List['TxSend'] = []
+
+    def __len__(self):
+        return len(self._messages)
+
+    def append(self, tx_item):
+        self._messages.append(tx_item.get_tx_message())
+        self._total_size += len(tx_item)
+
+    def get_messages(self) -> List['TxSend']:
+        return self._messages
+
+    def size(self) -> int:
+        return self._total_size
+
+
+class TxMessagesQueue(queue.Queue):
+    def __init__(self, maxsize=0, max_tx_size=None, max_tx_count=None):
+        super().__init__(maxsize=maxsize)
```

Comment:
`maxsize` of `queue.Queue` may lead the thread stuck at `self.put`.

Suppose that `maxsize` is supplied to `TxMessagesQueue`. And it reaches to its `maxsize` by appending tx_items.

In this case, `queue.Queue.put` will be blocked until the item of `queue.Queue` is consumed.
But this blocking situation cannot be resolved because `append` has locked and no other threads can `get` to consume the queue. 

This means that the running thread is blocked forever.